sdk/communication/Azure.Communication.CallAutomation/src/Models/Events/ReasonCodeName.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ComponentModel;
+using System;
+
+namespace Azure.Communication.CallAutomation
+{
+    /// <summary>
+    /// Failure Reason for incoming webhook events.
+    /// </summary>
+    public readonly partial struct ReasonCodeName : IEquatable<ReasonCodeName>
+    {
+        private readonly string _value;
+
+        /// <summary> Initializes a new instance of <see cref="ReasonCodeName"/>. </summary>
+        /// <exception cref="ArgumentNullException"> <paramref name="value"/> is null. </exception>
+        public ReasonCodeName(string value)
+        {
+            _value = value ?? throw new ArgumentNullException(nameof(value));
+        }
+
+        private const string RecognizeInitialSilenceTimeoutValue = "8510";
+        private const string RecognizeInterDigitTimeoutValue = "8532";
+        private const string RecognizePlayPromptFailedValue = "8511";
+
+        private const string RecognizeMaxDigitsReceivedValue = "8531";
+        private const string RecognizeStopToneDetectedValue = "8514";
+
+        private const string PlayDownloadFailedValue = "8536";
+        private const string PlayInvalidFileFormatValue = "8535";
+
+        private const string CompletedSuccessfullyValue = "0";
+        private const string OperationCanceledValue = "8508";
+        private const string UspecifiedErrorValue = "9999";
+
+        /// <summary> Action failed, initial silence timeout reached. </summary>
+        public static ReasonCodeName RecognizeInitialSilenceTimeout { get; } = new ReasonCodeName(RecognizeInitialSilenceTimeoutValue);
+        /// <summary> Action failed, inter-digit silence timeout reached. </summary>
+        public static ReasonCodeName RecognizeInterDigitTimeout { get; } = new ReasonCodeName(RecognizeInterDigitTimeoutValue);
```

Comment:
Sorry I missed this in last review should we name this to `RecognizeInterDigitTimedOut` ? I think they should be in past tense?  (and same goes for other timeout enum)